routing/pathfind.go

```diff
@@ -718,6 +721,12 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 			inboundFee = edge.inboundFees.CalcFee(
 				toNodeDist.netAmountReceived,
 			)
+
+			// Make sure that the node total fee is never negative.
```

Comment:
nit: perhaps we could add that this also limits the way a routing node can attract payments (in addition to the channel fee check)

Reply:
Extended comment. But I think the reason now isn't really limiting honey pot patterns, but merely just treating a negative fee in line with what routing nodes do with it (treat as zero fee)?